Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -733,6 +735,31 @@ void CanvasRenderingContext2DBase::setGlobalCompositeOperation(const String& ope
     c->setCompositeOperation(op, blendMode);
 }
 
+void CanvasRenderingContext2DBase::setFilterString(const String& filterString)
+{
+    if (state().filterString == filterString)
+        return;
+
+    // Spec: context.filter = "" should leave the current filter unchanged.
+    if (filterString.isEmpty())
+        return;
+
+    // Spec: context.filter = null or context.filter = undefined should leave the current filter unchanged.
+    if (filterString == "null"_s || filterString == "undefined"_s)
```

Comment:
I don't think you'll get here with a literal "null" or "undefined" string here in the cases the spec is talking about. It would be something like:

```
let foo;
context.filter = foo;
```

Reply:
When the filterString is passed as `foo`, JS binding sends "undefined" string. I added this case to the layout test.

This if-statement was kind of an optimization. I wanted to avoid calling `createFilterOperations()` try to see if the string is parsable or not. But I removed it.